library/src/actions/isoCountryCode/isoCountryCode.ts

```diff
@@ -0,0 +1,401 @@
+  { name: 'Viet Nam', alphaTwo: 'VN', alphaThree: 'VNM' },
+  { name: 'Virgin Islands (British)', alphaTwo: 'VG', alphaThree: 'VGB' },
+  { name: 'Virgin Islands (U.S.)', alphaTwo: 'VI', alphaThree: 'VIR' },
+  { name: 'Wallis and Futuna', alphaTwo: 'WF', alphaThree: 'WLF' },
+  { name: 'Western Sahara', alphaTwo: 'EH', alphaThree: 'ESH' },
+  { name: 'Yemen', alphaTwo: 'YE', alphaThree: 'YEM' },
+  { name: 'Zambia', alphaTwo: 'ZM', alphaThree: 'ZMB' },
+  { name: 'Zimbabwe', alphaTwo: 'ZW', alphaThree: 'ZWE' },
+] as const;
+
+/**
+ * ISO 3166-1 country code issue interface.
+ */
+export interface IsoCountryCodeIssue<TInput extends string>
+  extends BaseIssue<TInput> {
+  /**
+   * The issue kind.
+   */
+  readonly kind: 'validation';
+  /**
+   * The issue type.
+   */
+  readonly type: 'iso_country_code';
+  /**
+   * The expected property.
+   */
+  readonly expected: null;
+  /**
+   * The received property.
+   */
+  readonly received: `"${string}"`;
+  /**
+   * The validation function.
+   */
+  readonly requirement: (input: string) => boolean;
+}
+
+/**
+ * ISO 3166-1 country code action interface.
+ */
+export interface IsoCountryCodeAction<
+  TInput extends string,
+  TMessage extends ErrorMessage<IsoCountryCodeIssue<TInput>> | undefined,
+> extends BaseValidation<TInput, TInput, IsoCountryCodeIssue<TInput>> {
+  /**
+   * The action type.
+   */
+  readonly type: 'iso_country_code';
+  /**
+   * The action reference.
+   */
+  readonly reference: typeof isoCountryCode;
+  /**
+   * The expected property.
+   */
+  readonly expects: null;
+  /**
+   * The validation function.
+   */
+  readonly requirement: (input: string) => boolean;
+  /**
+   * The error message.
+   */
+  readonly message: TMessage;
+}
+
+/**
+ * Creates an [ISO 3166-1](https://en.wikipedia.org/wiki/ISO_3166-1) country code action.
+ *
+ * @returns An ISO 3166-1 country code action.
+ *
+ * @beta
+ */
+export function isoCountryCode<TInput extends string>(): IsoCountryCodeAction<
+  TInput,
+  undefined
+>;
+
+/**
+ * Creates an [ISO 3166-1](https://en.wikipedia.org/wiki/ISO_3166-1) country code action.
+ *
+ * @param message The error message.
+ *
+ * @returns An ISO 3166-1 country code action.
+ *
+ * @beta
+ */
+export function isoCountryCode<
+  TInput extends string,
+  const TMessage extends ErrorMessage<IsoCountryCodeIssue<TInput>> | undefined,
+>(message: TMessage): IsoCountryCodeAction<TInput, TMessage>;
+
+// @__NO_SIDE_EFFECTS__
+export function isoCountryCode(
+  message?: ErrorMessage<IsoCountryCodeIssue<string>>
+): IsoCountryCodeAction<
+  string,
+  ErrorMessage<IsoCountryCodeIssue<string>> | undefined
+> {
+  return {
+    kind: 'validation',
+    type: 'iso_country_code',
+    reference: isoCountryCode,
+    async: false,
+    expects: null,
+    requirement(input) {
+      return ISO_3166_1_ENTRIES.some(
+        (entry) => entry.alphaTwo === input || entry.alphaThree === input
```

Comment:
My opinion: each check should be a separate action, since some people only need one thing, while others need everything. If someone doesn't need the three-letter check, they'll get an extra code in their application.

ISO 3166-1-alpha-2
ISO 3166-1-alpha-3
ISO 3166-1-numeric

Plus, in the current version, it's unclear why we don't check the numeric value, even though it's part of the standard.